### PR TITLE
fix: External URL checks by mocking proposals

### PR DESCRIPTION
**Pull request by NabinKawan**
## List of changes

- Fixed test 5G and 5H

## Checklist

- [related issue](https://github.com/IntersectMBO/govtool/issues/)
- [x] My changes generate no new warnings
- [x] My code follows the [style guidelines](https://github.com/IntersectMBO/govtool/tree/main/docs/style-guides) of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the [changelog](https://github.com/IntersectMBO/govtool/blob/main/CHANGELOG.md)
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: tests/govtool-frontend/playwright/tests/5-proposal-functionality/proposalFunctionality.dRep.spec.ts [hunk 1]
```diff
@@ -85,34 +83,48 @@ test.describe("Proposal checks", () => {
...
-    } else {
-      expect(true, "Is not a bad or data missing proposal").toBeTruthy();
-    }
+    const govActionDetailsPage = await govActionsPage.viewFirstProposal();
+
+    await govActionDetailsPage.externalModalBtn.click();
+    await govActionDetailsPage.continueModalBtn.click();
+    const existingPages = page.context().pages();
+    expect(existingPages).toHaveLength(1);
+    expect(true, "Is not a bad or data missing proposal").toBeTruthy();
```
Comment (mesudip) [inline tests/govtool-frontend/playwright/tests/5-proposal-functionality/proposalFunctionality.dRep.spec.ts:127]: Remove this

File: tests/govtool-frontend/playwright/tests/5-proposal-functionality/proposalFunctionality.dRep.spec.ts [hunk 2]
```diff
@@ -85,34 +83,48 @@ test.describe("Proposal checks", () => {
...
+
+    await govActionDetailsPage.externalModalBtn.click();
+
+    await expect(govActionDetailsPage.externalLinkModal).toBeVisible();
+    await expect(
+      govActionDetailsPage.currentPage.getByText("Be careful", {
+        exact: false,
+      })
+    ).toBeVisible();
+    expect(true, "Is not a bad or data missing proposal").toBeTruthy();
```
Comment (mesudip) [inline tests/govtool-frontend/playwright/tests/5-proposal-functionality/proposalFunctionality.dRep.spec.ts:115]: Remove this unnecessary assertion

**Review by mesudip — CHANGES_REQUESTED** [review]

File: tests/govtool-frontend/playwright/tests/5-proposal-functionality/proposalFunctionality.dRep.spec.ts (hunk 1 above)
Comment (NabinKawan) [reply tests/govtool-frontend/playwright/tests/5-proposal-functionality/proposalFunctionality.dRep.spec.ts]: Thanks, It has been removed.

File: tests/govtool-frontend/playwright/tests/5-proposal-functionality/proposalFunctionality.dRep.spec.ts (hunk 2 above)
Comment (NabinKawan) [reply tests/govtool-frontend/playwright/tests/5-proposal-functionality/proposalFunctionality.dRep.spec.ts]: Thanks, It has been removed.

**Review by mesudip — APPROVED** [review]

**Pull request merged**